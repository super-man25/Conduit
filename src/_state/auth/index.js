export { default as saga } from './saga';

const reducerName = 'auth';
const createActionType = (type) => `${reducerName}/${type}`;

// Initial State / Reducer
export const initialState = {
  model: null,
  loading: true
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_USER:
      return { ...state, model: action.payload };
    case UNSET_USER:
      return { ...state, model: null };
    case IS_PENDING:
      return { ...state, loading: action.payload };
    default:
      return state;
  }
};

// Selectors
const selectUser = (store) => store[reducerName].model;
const selectLoading = (store) => store[reducerName].loading;

// Action Types
const FETCH_ASYNC = createActionType('FETCH_ASYNC');
const IS_PENDING = createActionType('IS_PENDING');
const SET_USER = createActionType('SET_USER');
const UNSET_USER = createActionType('UNSET_USER');
const SIGN_IN_ASYNC = createActionType('SIGN_IN_ASYNC');
const SIGN_OUT_ASYNC = createActionType('SIGN_OUT_ASYNC');
const FORGOT_PASS_ASYNC = createActionType('FORGOT_PASS_ASYNC');

// Actions
const signIn = (email, password) => ({
  type: SIGN_IN_ASYNC,
  payload: { email, password }
});
const signOut = () => ({ type: SIGN_OUT_ASYNC });
const forgotPass = (email) => ({ type: FORGOT_PASS_ASYNC, payload: { email } });
const fetch = () => ({ type: FETCH_ASYNC });

// Exports
export const types = {
  FETCH_ASYNC,
  IS_PENDING,
  SET_USER,
  UNSET_USER,
  SIGN_IN_ASYNC,
  SIGN_OUT_ASYNC,
  FORGOT_PASS_ASYNC
};
export const actions = {
  signIn,
  signOut,
  forgotPass,
  fetch
};
export const selectors = {
  selectUser,
  selectLoading
};
