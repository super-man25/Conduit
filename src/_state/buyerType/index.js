// @flow
import type { EDBuyerType } from '_models/buyerType';

export { default as saga } from './saga';

const FETCH_BUYER_TYPES = 'buyerType/FETCH_BUYER_TYPES';
const FETCH_BUYER_TYPES_SUCCESS = 'buyerType/FETCH_BUYER_TYPES_SUCCESS';
const FETCH_BUYER_TYPES_ERROR = 'buyerType/FETCH_BUYER_TYPES_ERROR';
const UPDATE_BUYER_TYPES = 'buyerType/UPDATE_BUYER_TYPES';
const UPDATE_BUYER_TYPES_SUCCESS = 'buyerType/UPDATE_BUYER_TYPES_SUCCESS';
const UPDATE_BUYER_TYPES_ERROR = 'buyerType/UPDATE_BUYER_TYPES_ERROR';
const RESET = 'buyerType/RESET';

// Action types
export type FetchBuyerTypesAction = {
  type: 'buyerType/FETCH_BUYER_TYPES',
  payload: { seasonId: number }
};

export type FetchBuyerTypesSuccessAction = {
  type: 'buyerType/FETCH_BUYER_TYPES_SUCCESS',
  payload: EDBuyerType[]
};

export type FetchBuyerTypesErrorAction = {
  type: 'buyerType/FETCH_BUYER_TYPES_ERROR',
  payload: Error
};

export type UpdateBuyerTypesAction = {
  type: 'buyerType/UPDATE_BUYER_TYPES',
  payload: any
};

export type UpdateBuyerTypesSuccessAction = {
  type: 'buyerType/UPDATE_BUYER_TYPES_SUCCESS',
  payload: number
};

export type UpdateBuyerTypeErrorAction = {
  type: 'buyerType/UPDATE_BUYER_TYPES_ERROR',
  payload: Error
};

export type ResetBuyerTypesAction = {
  type: 'buyerType/RESET'
};

export type Action =
  | FetchBuyerTypesAction
  | FetchBuyerTypesSuccessAction
  | FetchBuyerTypesErrorAction
  | UpdateBuyerTypesAction
  | UpdateBuyerTypesSuccessAction
  | UpdateBuyerTypeErrorAction
  | ResetBuyerTypesAction;

export const types = {
  FETCH_BUYER_TYPES,
  FETCH_BUYER_TYPES_SUCCESS,
  FETCH_BUYER_TYPES_ERROR,
  UPDATE_BUYER_TYPES,
  UPDATE_BUYER_TYPES_SUCCESS,
  UPDATE_BUYER_TYPES_ERROR,
  RESET
};

// Actions
const fetchBuyerTypes = (payload: {
  seasonId: number
}): FetchBuyerTypesAction => ({
  type: FETCH_BUYER_TYPES,
  payload
});

const updateBuyerTypes = (
  buyerTypes: EDBuyerType[]
): UpdateBuyerTypesAction => ({
  type: UPDATE_BUYER_TYPES,
  payload: buyerTypes
});

const resetBuyerTypes = (): ResetBuyerTypesAction => ({ type: RESET });

export const actions = {
  fetchBuyerTypes,
  updateBuyerTypes,
  resetBuyerTypes
};

// State/reducer
type State = {
  buyerTypes: EDBuyerType[],
  loading: boolean,
  error: ?Error
};

export const initialState: State = {
  buyerTypes: [],
  loading: false,
  error: null
};

type Store = {
  buyerType: State
};

export const reducer = (state: State = initialState, action: Action) => {
  switch (action.type) {
    case FETCH_BUYER_TYPES:
      return { ...state, error: null, loading: true };
    case FETCH_BUYER_TYPES_SUCCESS:
      return {
        ...state,
        loading: false,
        error: null,
        buyerTypes: action.payload
      };
    case FETCH_BUYER_TYPES_ERROR:
      return { ...state, loading: false, error: action.payload };
    case UPDATE_BUYER_TYPES:
      return { ...state, loading: true };
    case UPDATE_BUYER_TYPES_SUCCESS:
      return {
        ...state,
        loading: false,
        error: null
      };
    case UPDATE_BUYER_TYPES_ERROR:
      return { ...state, error: action.payload, loading: false };
    case RESET:
      return initialState;
    default:
      return state;
  }
};

// selectors
const selectAllBuyerTypes = (store: Store) => store.buyerType.buyerTypes;
const selectIsLoading = (store: Store) => store.buyerType.loading;
const selectError = (store: Store) => store.buyerType.error;

export const selectors = {
  selectAllBuyerTypes,
  selectIsLoading,
  selectError
};
