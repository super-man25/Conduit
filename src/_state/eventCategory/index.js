// @flow
import type { EDEventCategory } from '_models/eventCategory';

export { default as saga } from './saga';

const FETCH_EVENT_CATEGORIES = 'eventCategory/FETCH_EVENT_CATEGORIES';
const FETCH_EVENT_CATEGORIES_SUCCESS =
  'eventCategory/FETCH_EVENT_CATEGORIES_SUCCESS';
const FETCH_EVENT_CATEGORIES_ERROR =
  'eventCategory/FETCH_EVENT_CATEGORIES_ERROR';
const RESET_EVENT_CATEGORIES = 'eventCategory/RESET_EVENT_CATEGORIES';

// Action types
export type FetchEventCategoryAction = {
  type: 'eventCategory/FETCH_EVENT_CATEGORIES'
};

export type FetchEventCategorySuccessAction = {
  type: 'eventCategory/FETCH_EVENT_CATEGORIES_SUCCESS',
  payload: EDEventCategory[]
};

export type FetchEventCategoryErrorAction = {
  type: 'eventCategory/FETCH_EVENT_CATEGORIES_ERROR',
  payload: Error
};

export type ResetEventCategoryAction = {
  type: 'eventCategory/RESET_EVENT_CATEGORIES'
};

export type Action =
  | FetchEventCategoryAction
  | FetchEventCategorySuccessAction
  | FetchEventCategoryErrorAction
  | ResetEventCategoryAction;

export const types = {
  FETCH_EVENT_CATEGORIES,
  FETCH_EVENT_CATEGORIES_SUCCESS,
  FETCH_EVENT_CATEGORIES_ERROR,
  RESET_EVENT_CATEGORIES
};

// Actions
const fetchEventCategories = (): FetchEventCategoryAction => ({
  type: FETCH_EVENT_CATEGORIES
});

const resetEventCategories = (): ResetEventCategoryAction => ({
  type: RESET_EVENT_CATEGORIES
});

export const actions = {
  fetchEventCategories,
  resetEventCategories
};

// State/reducer
type State = {
  eventCategories: EDEventCategory[],
  loading: boolean,
  error: ?Error
};

export const initialState: State = {
  eventCategories: [],
  loading: false,
  error: null
};

type Store = {
  eventCategory: State
};

export const reducer = (state: State = initialState, action: Action) => {
  switch (action.type) {
    case FETCH_EVENT_CATEGORIES:
      return { ...state, loading: true };
    case FETCH_EVENT_CATEGORIES_SUCCESS:
      return { ...state, loading: false, eventCategories: action.payload };
    case FETCH_EVENT_CATEGORIES_ERROR:
      return { ...state, loading: false, error: action.payload };
    case RESET_EVENT_CATEGORIES:
      return { ...initialState };
    default:
      return state;
  }
};

// selectors
const selectAllEventCategories = (store: Store) =>
  store.eventCategory.eventCategories;

const selectEventCategoryMap = (store: Store) =>
  store.eventCategory.eventCategories.reduce((acc, eventCategory) => {
    const { id, ...rest } = eventCategory;
    return { ...acc, [id]: rest };
  }, {});

const selectIsLoading = (store: Store) => store.eventCategory.loading;

export const selectors = {
  selectAllEventCategories,
  selectEventCategoryMap,
  selectIsLoading
};
