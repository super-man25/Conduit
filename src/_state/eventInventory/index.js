// @flow
import { calculateFilteredRows } from './utils';
import { createSelector } from 'reselect';
import type { EDInventoryRow } from '_models';
export { default as saga } from './saga';

// Action Types
const FETCH_EVENT_INVENTORY = 'eventInventory/FETCH_EVENT_INVENTORY';
const FETCH_EVENT_INVENTORY_SUCCESS =
  'eventInventory/FETCH_EVENT_INVENTORY_SUCCESS';
const FETCH_EVENT_INVENTORY_ERROR =
  'eventInventory/FETCH_EVENT_INVENTORY_ERROR';
const SET_EVENT_INVENTORY_FILTER = 'eventInventory/SET_EVENT_INVENTORY_FILTER';
const RESET = 'eventInventory/RESET';
const SET_EVENT_ROW_LISTED_REQUEST =
  'eventInventory/SET_EVENT_ROW_LISTED_REQUEST';
const SET_EVENT_ROW_LISTED_SUCCESS =
  'eventInventory/SET_EVENT_ROW_LISTED_SUCCESS';
const SET_EVENT_ROW_LISTED_ERROR = 'eventInventory/SET_EVENT_ROW_LISTED_ERROR';
const SET_EVENT_ROW_MANUAL_PRICE_REQUEST =
  'eventInventory/SET_EVENT_ROW_MANUAL_PRICE_REQUEST';
const SET_EVENT_ROW_MANUAL_PRICE_SUCCESS =
  'eventInventory/SET_EVENT_ROW_MANUAL_PRICE_SUCCESS';
const SET_EVENT_ROW_MANUAL_PRICE_ERROR =
  'eventInventory/SET_EVENT_ROW_MANUAL_PRICE_ERROR';
const SELECT_EVENT_ROW = 'eventInventory/SELECT_EVENT_ROW';
const SELECT_ALL_EVENT_ROWS = 'eventInventory/SELECT_ALL_EVENT_ROWS';
const SET_EDITING_MANUAL_PRICE = 'eventInventory/SET_EDITING_MANUAL_PRICE';
const CANCEL_EDITING_MANUAL_PRICE =
  'eventInventory/CANCEL_EDITING_MANUAL_PRICE';
const SET_SCALE_FILTERS = 'eventInventory/SET_SCALE_FILTERS';
const CLEAR_SELECTED_SCALE_FILTERS =
  'eventInventory/CLEAR_SELECTED_SCALE_FILTERS';
const SET_SELECTED_SCALE_FILTERS = 'eventInventory/SET_SELECTED_SCALE_FILTERS';

export type FetchEventInventoryAction = {
  type: 'eventInventory/FETCH_EVENT_INVENTORY',
  payload: number
};
export type FetchEventInventorySuccessAction = {
  type: 'eventInventory/FETCH_EVENT_INVENTORY_SUCCESS',
  payload: any
};
export type FetchEventInventoryErrorAction = {
  type: 'eventInventory/FETCH_EVENT_INVENTORY_ERROR',
  payload: Error
};
export type ResetEventInventoryAction = { type: 'eventInventory/RESET' };
export type SetEventInventoryFilterAction = {
  type: 'eventInventory/SET_EVENT_INVENTORY_FILTER',
  payload: string
};
export type SetEventRowListedRequestAction = {
  type: 'eventInventory/SET_EVENT_ROW_LISTED_REQUEST',
  payload: { row: EDInventoryRow, value: boolean }
};
export type SetEventRowListedSuccessAction = {
  type: 'eventInventory/SET_EVENT_ROW_LISTED_SUCCESS',
  payload: { row: EDInventoryRow, value: boolean }
};
export type SetEventRowListedErrorAction = {
  type: 'eventInventory/SET_EVENT_ROW_LISTED_ERROR',
  payload: { row: EDInventoryRow, value: boolean }
};
export type SetEventRowManualPriceRequestAction = {
  type: 'eventInventory/SET_EVENT_ROW_MANUAL_PRICE_REQUEST',
  payload: { row: EDInventoryRow, value: string }
};
export type SetEventRowManualPriceSuccessAction = {
  type: 'eventInventory/SET_EVENT_ROW_MANUAL_PRICE_SUCCESS',
  payload: { row: EDInventoryRow }
};
export type SetEventRowManualPriceErrorAction = {
  type: 'eventInventory/SET_EVENT_ROW_MANUAL_PRICE_ERROR',
  payload: { row: EDInventoryRow, error: Error }
};
export type SelectEventRowAction = {
  type: 'eventInventory/SELECT_EVENT_ROW',
  payload: number
};
export type SelectAllEventRowsAction = {
  type: 'eventInventory/SELECT_ALL_EVENT_ROWS'
};
export type SetEditingManualPriceAction = {
  type: 'eventInventory/SET_EDITING_MANUAL_PRICE',
  payload: number
};
export type CancelEditingManualPriceAction = {
  type: 'eventInventory/CANCEL_EDITING_MANUAL_PRICE'
};
export type SetScaleFiltersAction = {
  type: 'eventInventory/SET_SCALE_FILTERS',
  payload: any[]
};
export type ClearSelectedScaleFiltersAction = {
  type: 'eventInventory/CLEAR_SELECTED_SCALE_FILTERS'
};
export type SetSelectedScaleFiltersAction = {
  type: 'eventInventory/SET_SELECTED_SCALE_FILTERS',
  payload: any[]
};

export type Action =
  | FetchEventInventoryAction
  | FetchEventInventorySuccessAction
  | FetchEventInventoryErrorAction
  | ResetEventInventoryAction
  | SetEventInventoryFilterAction
  | SetEventRowListedRequestAction
  | SetEventRowListedErrorAction
  | SetEventRowManualPriceRequestAction
  | SetEventRowManualPriceErrorAction
  | SelectEventRowAction
  | SelectAllEventRowsAction
  | SetEditingManualPriceAction
  | SetEventRowListedSuccessAction
  | CancelEditingManualPriceAction
  | SetSelectedScaleFiltersAction
  | ClearSelectedScaleFiltersAction
  | SetScaleFiltersAction;

export const types = {
  FETCH_EVENT_INVENTORY,
  FETCH_EVENT_INVENTORY_SUCCESS,
  FETCH_EVENT_INVENTORY_ERROR,
  SET_EVENT_INVENTORY_FILTER,
  RESET,
  SET_EVENT_ROW_LISTED_REQUEST,
  SET_EVENT_ROW_LISTED_SUCCESS,
  SET_EVENT_ROW_LISTED_ERROR,
  SET_EVENT_ROW_MANUAL_PRICE_REQUEST,
  SET_EVENT_ROW_MANUAL_PRICE_SUCCESS,
  SET_EVENT_ROW_MANUAL_PRICE_ERROR,
  SET_EDITING_MANUAL_PRICE,
  SELECT_EVENT_ROW,
  SELECT_ALL_EVENT_ROWS,
  CANCEL_EDITING_MANUAL_PRICE,
  SET_SCALE_FILTERS,
  SET_SELECTED_SCALE_FILTERS,
  CLEAR_SELECTED_SCALE_FILTERS
};

// Actions
const fetchEventInventory = (id: number): FetchEventInventoryAction => ({
  type: FETCH_EVENT_INVENTORY,
  payload: id
});

const resetEventInventory = (): ResetEventInventoryAction => ({ type: RESET });

const setEventInventoryFilter = (
  name: string
): SetEventInventoryFilterAction => ({
  type: SET_EVENT_INVENTORY_FILTER,
  payload: name
});

const setEventRowListed = (
  row: EDInventoryRow,
  value: boolean
): SetEventRowListedRequestAction => ({
  type: SET_EVENT_ROW_LISTED_REQUEST,
  payload: { row, value }
});

const setEventRowManualPrice = (
  row: EDInventoryRow,
  value: string
): SetEventRowManualPriceRequestAction => ({
  type: SET_EVENT_ROW_MANUAL_PRICE_REQUEST,
  payload: { row, value }
});

const selectEventRow = (id: number): SelectEventRowAction => ({
  type: SELECT_EVENT_ROW,
  payload: id
});

const selectAllEventRows = (): SelectAllEventRowsAction => ({
  type: SELECT_ALL_EVENT_ROWS
});

const setEditingManualPrice = (id: number): SetEditingManualPriceAction => ({
  type: SET_EDITING_MANUAL_PRICE,
  payload: id
});

const cancelEditingManualPrice = (): CancelEditingManualPriceAction => ({
  type: CANCEL_EDITING_MANUAL_PRICE
});

const setScaleFilters = (filters: any[]): SetScaleFiltersAction => ({
  type: SET_SCALE_FILTERS,
  payload: filters
});

const clearSelectedScaleFilters = (): ClearSelectedScaleFiltersAction => ({
  type: CLEAR_SELECTED_SCALE_FILTERS
});

const setSelectedScaleFilters = (
  filters: any[]
): SetSelectedScaleFiltersAction => ({
  type: SET_SELECTED_SCALE_FILTERS,
  payload: filters
});

export const actions = {
  fetchEventInventory,
  resetEventInventory,
  setEventInventoryFilter,
  setEventRowListed,
  setEventRowManualPrice,
  selectEventRow,
  selectAllEventRows,
  setEditingManualPrice,
  cancelEditingManualPrice,
  setScaleFilters,
  clearSelectedScaleFilters,
  setSelectedScaleFilters
};

// State/Reducer
type State = {
  allRows: EDInventoryRow[],
  loading: boolean,
  error: ?Error,
  filterDirection: 'asc' | 'desc',
  filterName: string,
  manualPriceEditId: ?number,
  selectedRowIds: number[],
  scaleFilters: number[],
  selectedScaleFilters: number[]
};

export const initialState: State = {
  allRows: [],
  loading: false,
  error: null,
  filterDirection: 'asc',
  filterName: '',
  manualPriceEditId: null,
  selectedRowIds: [],
  scaleFilters: [],
  selectedScaleFilters: []
};

export const reducer = (state: State = initialState, action: Action) => {
  switch (action.type) {
    case FETCH_EVENT_INVENTORY:
      return { ...state, loading: true };
    case FETCH_EVENT_INVENTORY_SUCCESS:
      return {
        ...state,
        allRows: action.payload,
        loading: false
      };
    case FETCH_EVENT_INVENTORY_ERROR:
      return { ...state, error: action.payload, loading: false };
    case SET_EVENT_INVENTORY_FILTER: {
      const filterName = action.payload;
      let filterDirection;

      if (filterName !== state.filterName) {
        filterDirection = 'asc';
      } else {
        filterDirection = state.filterDirection === 'asc' ? 'desc' : 'asc';
      }

      return {
        ...state,
        filterName,
        filterDirection
      };
    }
    case SELECT_EVENT_ROW: {
      const id = action.payload;
      const selectedRowIds: number[] = state.selectedRowIds.slice();
      const indexOfRowId = selectedRowIds.indexOf(id);

      if (indexOfRowId !== -1) {
        selectedRowIds.splice(indexOfRowId, 1);
      } else {
        selectedRowIds.push(id);
      }

      return {
        ...state,
        selectedRowIds
      };
    }
    case SELECT_ALL_EVENT_ROWS: {
      const rows = calculateFilteredRows(
        state.allRows,
        state.filterDirection,
        state.filterName,
        state.selectedScaleFilters
      );

      const selectedRowIds: number[] =
        rows.length === state.selectedRowIds.length
          ? []
          : rows.map((row) => row.id);

      return {
        ...state,
        selectedRowIds
      };
    }
    case SET_EVENT_ROW_LISTED_REQUEST: {
      const {
        row: { id },
        value
      } = action.payload;

      const allRows: EDInventoryRow[] = state.allRows.map(
        (row) => (row.id === id ? { ...row, isListed: value } : row)
      );

      return {
        ...state,
        allRows
      };
    }
    case SET_EVENT_ROW_LISTED_ERROR: {
      const { row: fallbackRow } = action.payload;
      const id = fallbackRow.id;

      const allRows: EDInventoryRow[] = state.allRows.map(
        (row) => (row.id === id ? { ...fallbackRow } : row)
      );

      return { ...state, allRows };
    }
    case SET_EDITING_MANUAL_PRICE: {
      return { ...state, manualPriceEditId: action.payload };
    }
    case CANCEL_EDITING_MANUAL_PRICE: {
      return { ...state, manualPriceEditId: null };
    }
    case SET_EVENT_ROW_MANUAL_PRICE_REQUEST: {
      const {
        row: { id },
        value
      } = action.payload;

      const allRows: EDInventoryRow[] = state.allRows.map(
        (row) =>
          row.id === id
            ? { ...row, overridePrice: value === '' ? null : Number(value) }
            : row
      );

      return {
        ...state,
        allRows,
        manualPriceEditId: null
      };
    }
    case SET_EVENT_ROW_MANUAL_PRICE_ERROR: {
      const { row: fallbackRow } = action.payload;
      const id = fallbackRow.id;

      const allRows: EDInventoryRow[] = state.allRows.map(
        (row) => (row.id === id ? { ...fallbackRow } : row)
      );

      return { ...state, allRows };
    }
    case SET_SCALE_FILTERS: {
      return {
        ...state,
        scaleFilters: action.payload,
        selectedRowIds: []
      };
    }
    case CLEAR_SELECTED_SCALE_FILTERS: {
      return {
        ...state,
        selectedScaleFilters: [],
        selectedRowIds: []
      };
    }
    case SET_SELECTED_SCALE_FILTERS: {
      return {
        ...state,
        selectedScaleFilters: action.payload,
        selectedRowIds: []
      };
    }
    case RESET:
      return initialState;
    default:
      return state;
  }
};

// Selectors
type Store = {
  eventInventory: State
};

const selectAllEventInventoryRows = (store: Store) =>
  store.eventInventory.allRows;

const selectSelectedScaleFilters = (store: Store) =>
  store.eventInventory.selectedScaleFilters;

const selectEventInventoryLoading = (store: Store) =>
  store.eventInventory.loading;

const selectEventInventoryError = (store: Store) => store.eventInventory.error;

const selectSelectedRowIds = (store: Store) =>
  store.eventInventory.selectedRowIds;

const selectSelectedRows = createSelector(
  [selectAllEventInventoryRows, selectSelectedRowIds],
  (rows, ids) => {
    return ids.map((id) => rows.find((row) => row.id === id));
  }
);

const selectEventInventoryFilter = (store: Store) => {
  const { eventInventory } = store;
  return {
    name: eventInventory.filterName,
    direction: eventInventory.filterDirection
  };
};

const selectEventInventoryRows = createSelector(
  [
    selectAllEventInventoryRows,
    selectEventInventoryFilter,
    selectSelectedScaleFilters
  ],
  (rows, filters, selectedScaleFilters) =>
    calculateFilteredRows(
      rows,
      filters.direction,
      filters.name,
      selectedScaleFilters
    )
);

export const selectors = {
  selectAllEventInventoryRows,
  selectEventInventoryRows,
  selectEventInventoryLoading,
  selectEventInventoryError,
  selectEventInventoryFilter,
  selectSelectedRowIds,
  selectSelectedRows
};
