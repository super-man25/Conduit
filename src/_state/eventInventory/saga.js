// @flow
import { types, actions } from '.';
import {
  fork,
  take,
  put,
  cancel,
  takeLatest,
  call,
  select,
  all
} from 'redux-saga/effects';
import { eventService, venueService } from '_services';
import type { Saga } from 'redux-saga';
import { actions as alertActions } from '_state/alert';
import { selectors as eventSelectors } from '../event';
import type {
  FetchEventInventoryAction,
  SetEventRowListedRequestAction,
  SetEventRowManualPriceRequestAction
} from '.';
import {
  alphaFirstSort,
  findUniqueSections,
  mapSectionsToPriceScales
} from './utils';

export function* fetchEventInventory(
  action: FetchEventInventoryAction
): Saga<void> {
  try {
    const { id, venueId } = yield select(eventSelectors.selectEvent);

    const [eventInventory, priceScales] = yield all([
      call(eventService.getInventory, id),
      call(venueService.getPriceScales, venueId)
    ]);
    yield put(
      actions.setSectionsToPriceScaleAction(
        yield call(mapSectionsToPriceScales, eventInventory)
      )
    );
    const sortedPriceScales = alphaFirstSort(priceScales, 'name');

    yield put(actions.setScaleFilters(sortedPriceScales));
    yield put(
      actions.setSectionFilters(yield call(findUniqueSections, eventInventory))
    );
    yield put({
      type: types.FETCH_EVENT_INVENTORY_SUCCESS,
      payload: eventInventory
    });
    yield put({
      type: types.SET_EVENT_INVENTORY_FILTER,
      payload: 'priceScaleId'
    });
  } catch (err) {
    yield put({ type: types.FETCH_EVENT_INVENTORY_ERROR, payload: err });
  }
}

export function* setRowListed(
  action: SetEventRowListedRequestAction
): Saga<void> {
  const { row, value } = action.payload;
  const updateParams = { eventSeatIds: row.seats, isListed: value };

  try {
    yield call(eventService.updateEventSeats, updateParams);

    yield put({
      type: types.SET_EVENT_ROW_LISTED_SUCCESS,
      payload: action.payload
    });
  } catch (error) {
    yield put(
      alertActions.error(
        `Could not update section ${row.section} row ${row.row}.`
      )
    );
    yield put({
      type: types.SET_EVENT_ROW_LISTED_ERROR,
      payload: {
        error,
        row
      }
    });
  }
}

export function* setOverridePrice(
  action: SetEventRowManualPriceRequestAction
): Saga<void> {
  const { row, value } = action.payload;
  const updateParams = {
    eventSeatIds: row.seats,
    overridePrice: value.length ? value : null
  };

  try {
    yield call(eventService.updateEventSeats, updateParams);

    yield put({
      type: types.SET_EVENT_ROW_MANUAL_PRICE_SUCCESS,
      payload: action.payload
    });
  } catch (error) {
    let msg =
      error.name === 'ValidationError'
        ? error.errors[0]
        : `Could not update section ${row.section} row ${row.row}.`;

    yield put(alertActions.error(msg));
    yield put({
      type: types.SET_EVENT_ROW_MANUAL_PRICE_ERROR,
      payload: {
        error,
        row
      }
    });
  }
}

const takeLatestById = (pattern, saga, actionIdGetter, ...args) =>
  fork(function*() {
    const tasks = {};

    while (true) {
      const action = yield take(pattern);
      const id = actionIdGetter(action);

      if (tasks[id]) {
        yield cancel(tasks[id]);
      }

      tasks[id] = yield fork(saga, action);
    }
  });

function* watchFetchEventInventory(): Saga<void> {
  yield takeLatest(types.FETCH_EVENT_INVENTORY, fetchEventInventory);
}

function* watchSetRowListed(): Saga<void> {
  yield takeLatestById(
    types.SET_EVENT_ROW_LISTED_REQUEST,
    setRowListed,
    (action) => action.payload.row.id
  );
}

function* watchSetManualPrice(): Saga<void> {
  yield takeLatestById(
    types.SET_EVENT_ROW_MANUAL_PRICE_REQUEST,
    setOverridePrice,
    (action) => action.payload.row.id
  );
}

export default {
  watchFetchEventInventory,
  watchSetRowListed,
  watchSetManualPrice
};
