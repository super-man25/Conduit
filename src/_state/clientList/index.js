import reducer from './reducer';
import actions from './actions';
import saga from './saga';
import selectors from './selectors';

export { reducer, actions, saga, selectors };
