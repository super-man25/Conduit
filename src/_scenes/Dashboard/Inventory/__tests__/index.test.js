import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { EventInventory } from '../';
import { initialState } from '_state/eventInventory';

describe('<EventInventory /> index route', () => {
  const props = {
    event: {},
    selectedEventIds: [],
    isBulkUpdating: false,
    startBulkUpdate: jest.fn(),
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <EventInventory {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when there are selectedEventIds', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <EventInventory {...props} selectedEventIds={[1, 2, 3]} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when there are selectedEventIds and isBulkUpdating is true', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <EventInventory
          {...props}
          selectedEventIds={[1, 2, 3]}
          isBulkUpdating={true}
        />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
