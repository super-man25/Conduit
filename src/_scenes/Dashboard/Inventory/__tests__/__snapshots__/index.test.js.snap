// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventInventory /> index route should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <EventInventory
    event={Object {}}
    isBulkUpdating={false}
    selectedEventIds={Array []}
    startBulkUpdate={[MockFunction]}
  />
</ContextProvider>
`;

exports[`<EventInventory /> index route should render correctly when there are selectedEventIds 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <EventInventory
    event={Object {}}
    isBulkUpdating={false}
    selectedEventIds={
      Array [
        1,
        2,
        3,
      ]
    }
    startBulkUpdate={[MockFunction]}
  />
</ContextProvider>
`;

exports[`<EventInventory /> index route should render correctly when there are selectedEventIds and isBulkUpdating is true 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <EventInventory
    event={Object {}}
    isBulkUpdating={true}
    selectedEventIds={
      Array [
        1,
        2,
        3,
      ]
    }
    startBulkUpdate={[MockFunction]}
  />
</ContextProvider>
`;
